### PR TITLE
EgoNet

### LINKED ISSUE #475: [FEA] Ego graphs
**Is your feature request related to a problem? Please describe.**
It would be very helpful if `cugraph` could have the ability to calculate the ego graph to a provided radius, as is done in `networkx` via [https://networkx.github.io/documentation/stable/reference/generated/networkx.generators.ego.ego_graph.html?highlight=ego_graph#networkx.generators.ego.ego_graph](https://networkx.github.io/documentation/stable/reference/generated/networkx.generators.ego.ego_graph.html?highlight=ego_graph#networkx.generators.ego.ego_graph).

**Describe the solution you'd like**
Given a graph, the user would provide a series of required and optional values, just like `networkx.generators.ego.ego_graph`.  From there, the method returns a subgraph of neighbors, centered at the given node, within a given radius.

**Describe alternatives you've considered**
This can be done in `networkx` but is rather slow, particularly if you want to do it for multiple target nodes.  This would lend itself well to parallelization.

**Task List**
Unknown.

**Additional context**
None

### PR DESCRIPTION
**Pull request by afender**
### Description
Let the egonet graph of a node x be the subgraph that includes the neighborhood of x and all edges between them. Here is a basic description (1-hop, single seed) :
- Add center node x to the graph.
- Go through all the neighbors y of this center node x, add edge (x, y) to the graph. 
- For each neighbor y of center node x, go through all the neighbors z of center node x, if there is an edge between y and z in the original graph, add edge (y, z) to our new graph. 

### Proposed solution
Rather than doing custom one/two hops features, we propose a generic k-hops solution leveraging BFS with cutoff to identify neighbors within a given radius. 

In addition to the single source version (matching what's available in Nx), we propose to handle multiple sources (seeds) at once which allows better performances.

This PR also enables a path in the experimental stack for returning multiple graphs (edge list format) from CUDA prims to python without using the legacy classes.

As future work, we want to enable concurrency for the cutoff BFS for each seed. This is dependent of #957 

Close #475

### CODE REVIEW

**Review by Iroy30 — APPROVED** [review]

**Review by BradReesWork — APPROVED** [review]

File: cpp/include/utilities/cython.hpp [hunk 1]
```diff
@@ -109,6 +109,40 @@ struct graph_container_t {
   experimental::graph_properties_t graph_props;
 };
 
+/**
```
[the remark this answers is not in the document]
Comment (afender) [reply cpp/include/utilities/cython.hpp:113]: 98a71c5

File: cpp/include/algorithms.hpp [hunk 2]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 #pragma once
-
+#include <experimental/graph.hpp>
```
[the remark this answers is not in the document]
Comment (afender) [reply cpp/include/algorithms.hpp:17]: 0887b64

File: cpp/src/community/egonet.cu [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+#include "experimental/graph.hpp"
+#include "utilities/graph_utils.cuh"
+
+#include <experimental/graph_functions.hpp>
+#include <experimental/graph_view.hpp>
+
+namespace {
+/*
+Description
+Let the egonet graph of a node x be the subgraph that includes the neighborhood of x and all edges
```
[the remark this answers is not in the document]
Comment (afender) [reply cpp/src/community/egonet.cu]: Indeed, thanks for catching that.
 da23265

**Comment by BradReesWork** [comment]
@gpucibot merge

**Pull request merged**